cmd/goQuery/cmd/list.go

```diff
@@ -47,16 +50,37 @@ be broken up into IPv4 and IPv6 flows and the drops for that interface will be s
 }
 
 func listInterfacesEntrypoint(_ *cobra.Command, args []string) error {
-	return listInterfaces(viper.GetString(conf.QueryDBPath), args...)
+	return listInterfaces(context.Background(), viper.GetString(conf.QueryDBPath), args...)
 }
 
 // List interfaces for which data is available and show how many flows and
 // how much traffic was observed for each one.
-func listInterfaces(dbPath string, ifaces ...string) error {
+func listInterfaces(ctx context.Context, dbPath string, ifaces ...string) error {
 	queryArgs := cmdLineParams
 
 	// TODO: consider making this configurable
 	output := os.Stdout
+	var ifacesMetadata []*goDB.InterfaceMetadata
+
+	logger := logging.FromContext(ctx)
+
+	// create query logger
+	queryLogFile := viper.GetString(conf.QueryLog)
```

Comment:
`viper` in function calls is usually not something we should be going for. Can you make this an argument to the function?

```golang
func listInterfaces(ctx context.Context, dbPath, queryLog string, ifaces ...string) error { ... }
```